packages/validation/src/validators/admission-policy/validator.ts

```diff
@@ -112,34 +112,60 @@ export class AdmissionPolicyValidator extends AbstractPlugin {
     const {
       type,
       resource,
-      expression: {message, expression},
+      expression: {message, messageExpression, expression},
       level,
       params,
       property,
     } = args;
 
     let output: any;
+    let ruleMessage: string | undefined = message;
+    let messageExpressionOutput: string | undefined;
 
     if (type === 'validating-admission-policy') {
       output = (globalThis as any).eval(expression, YAML.stringify({object: resource.content, params})).output;
+
+      if (messageExpression) {
+        messageExpressionOutput = (globalThis as any).eval(
+          messageExpression,
+          YAML.stringify({object: resource.content, params})
+        ).output;
+
+        console.log(messageExpressionOutput);
+      }
     } else if (type === 'crd' && property) {
       output = (globalThis as any).eval(
         expression,
         YAML.stringify({self: property === '<root>' ? resource.content : resource.content[property]})
       ).output;
+
+      if (messageExpression) {
+        messageExpressionOutput = (globalThis as any).eval(
```

Comment:
The `(globalThis as any).eval(message, params).output` is used multiple times. I'd wrap a function `evaluateCelExpression` at the bottom of this file.